shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,134 @@
+// {
+//   @"location": 0,
+//   @"length" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+//
+// Returns nil if the language is invalid.
+// Returns an empty array if no spell check suggestions.
+- (NSArray<NSDictionary<NSString*, id>*>*)findAllSpellCheckSuggestionsForText:(NSString*)text
+                                                                   inLanguage:(NSString*)language {
+  if (![UITextChecker.availableLanguages containsObject:language]) {
+    return nil;
+  }
+
+  NSMutableArray<NSDictionary<NSString*, id>*>* allSpellSuggestions =
+      [[[NSMutableArray alloc] init] autorelease];
+
+  NSDictionary<NSString*, id>* nextSpellSuggestion;
+  NSInteger nextOffset = 0;
+  do {
+    nextSpellSuggestion = [self findSpellCheckSuggestionsForText:text
+                                                      inLanguage:language
+                                                  startingOffset:nextOffset];
+    if (nextSpellSuggestion != nil) {
+      [allSpellSuggestions addObject:nextSpellSuggestion];
+      nextOffset = [nextSpellSuggestion[@"location"] integerValue] +
+                   [nextSpellSuggestion[@"length"] integerValue];
+    }
+  } while (nextSpellSuggestion != nil);
+
+  return allSpellSuggestions;
+}
+
+// Get the misspelled word and suggestions.
+//
+// The format of result looks like:
+// {
+//   @"location": 0,
+//   @"length" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+//
+// Returns nil if no spell check suggestions.
+- (NSDictionary<NSString*, id>*)findSpellCheckSuggestionsForText:(NSString*)text
+                                                      inLanguage:(NSString*)language
+                                                  startingOffset:(NSInteger)startingOffset {
+  FML_DCHECK([UITextChecker.availableLanguages containsObject:language]);
+  NSRange misspelledRange =
+      [self.textChecker rangeOfMisspelledWordInString:text
+                                                range:NSMakeRange(0, text.length)
+                                           startingAt:startingOffset
+                                                 wrap:NO
+                                             language:language];
+  if (misspelledRange.location == NSNotFound) {
+    // No misspelled word found
+    return nil;
+  }
+
+  // If no possible guesses, the API returns an empty array:
+  // https://developer.apple.com/documentation/uikit/uitextchecker/1621037-guessesforwordrange?language=objc
+  NSArray<NSString*>* suggestions = [[self.textChecker guessesForWordRange:misspelledRange
+                                                                  inString:text
+                                                                  language:language] autorelease];
+  // Format the result.
+  NSMutableDictionary* result = [[[NSMutableDictionary alloc] initWithCapacity:3] autorelease];
+  result[@"location"] = @(misspelledRange.location);
+  result[@"length"] = @(misspelledRange.length);
```

Comment:
Instead of splitting the range you can use `NSValue`, see `valueWithRange:`.  Or better, create a new object with range and suggestions properties? Or a dictionary with the `NSValue` as a key and the suggestions.  The string keys needed between two methods seems unnecessary.